### ISSUE TITLE
missing build data for 45, 46, 47, 48, and 49

### ISSUE THREAD

**Issue by willkg** [mozilla-services/buildhub#528]
While working on Socorro's BetaVersionRule which figures out the real version for the build that an incoming crash report is reporting for, I discovered that there's a lot of build information missing on archive.mozilla.org and also on Buildhub.

For example, Buildhub has this for major version 48:

https://mozilla-services.github.io/buildhub/?q=48

Clearly there were more builds than nightlies and 48.0b1rc3.

If Buildhub is supposed to be a catalog of all build information, then this information should be found and added even if it's not on archive.mozilla.org.

This affects Socorro because we get crash reports for beta versions before Firefox 50 and if the BetaVersionRule relies solely on Buildhub information, it can't figure out the real versions. That makes switching to Buildhub as an information store less enthusing.

As a side note, Socorro has all this build information in the `product_versions` table. I'm not sure how it got it given that it's not on archive.mozilla.org.

**Comment by willkg** [comment]
Adjusting the summary. Buildhub is missing build data for anything less than Firefox 50.

**Comment by peterbe** [comment]
@leplatrem Can you help out here? Do you know why there might be certain versions missing in archive.mozilla.org?

Buildhub1, for all its admitted flaws, is a near reflection of the contents of the S3 buckets (`net-mozaws-prod-delivery-firefox` and `net-mozaws-prod-delivery-archive`). Basically, if a bunch of 45, 46, 47, 48, and 49 were once put into those S3 buckets but later removed, would Buildhub1 delete them too?

**Comment by leplatrem** [comment]
> would Buildhub1 delete them too?

We never delete anything in buildhub1

**Comment by willkg** [comment]
@peterbe and I talked with Nick and crew and what's going on is that they purge the `candidates/` directory periodically. So Socorro has data that isn't available anywhere else.

I think that probably means a couple of things:

1. I'm going to be using the scraper I wrote for the foreseeable future because I have the data I need and it's "good enough" for now and I need to spend time on other things.
2. If someone else needs this data, they can ping me and I can extract it. Until then, it's sitting in a dump file in Google Drive.

Given all that, I'm going to close this out because I don't think there's anything to do here.
